pkg/base1/cockpit.js

```diff
@@ -2894,19 +2911,38 @@ function full_scope(cockpit, $, po) {
         }
 
         self.fetch = function fetch(beg, end) {
-            var opts = $.extend({ }, self.options, {
-                timestamp: timestamp(beg, interval),
-                limit: end - beg
-            });
-            if (opts.source == "direct")
-                opts.source = "pcp-archive";
-            transfer(opts, drain);
+            /* Provide some fake data synchronously to the series so
+             * that it doesn't ask us again for the same range while
+             * we are still fetching it asynchronously.
+             *
+             * XXX - is this something the series should do itself?
+             */
+            self.series.input(beg, new Array(end-beg), { });
+
+            var timestamp = beg*interval - $.now();
+            var limit = end - beg;
+
+            var archive_options_list = [ ];
+            for (var i = 0; i < options_list.length; i++) {
+                if (options_list[i].archive_source) {
+                    archive_options_list.push($.extend({}, options_list[i],
+                                                       { "source": options_list[i].archive_source,
+                                                         timestamp: timestamp,
+                                                         limit: limit
+                                                       }));
+                }
+            }
+
+            transfer(archive_options_list, drain, true);
         };
 
+        var following = false;
+
         self.follow = function follow() {
-            var opts = $.extend({ }, self.options);
-            delete opts.limit;
-            transfer(opts, drain);
+            if (!following) {
+                following = true;
```

Comment:
It seems like following is never reset to false, when the following channel closes (for whatever reason).